circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/DestructiveReplication.java

```diff
@@ -52,6 +56,10 @@ public void replicate() throws CircusTrainException {
             + tableReplication.getQualifiedReplicaName()
             + "' is not controlled by circus train aborting replication, check configuration for correct replica name");
       }
+
+      hiveTableAnnotator.annotateTable(tableReplication.getReplicaDatabaseName(),
```

Comment:
Not sure this is needed? I'd say we only have to annotate when we create a new table.

Reply:
Yeah I'm not 100% sure about the annotation logic in this class. CT does `destructiveReplica.dropDeletedPartitions()` and `destructiveReplica.dropTable()` on the replica which are both then scheduled for housekeeping cleanup. In which case I thought it makes sense to annotate the table _before_ peforming these actions so the paths are cleaned up.

But now that the table annotations aren't strictly about housekeeping/beekeeper, I think you're right we should only be applying them when we create a new table.

I think this will mean that when a user runs a destructive replication, they will need to annotate their table manually first? (otherwise these paths will be missed)

Reply:
I guess there are 2 usecases:
1) new table is being replicated and we want to use beekeeper
2) existing table is already being replicated and we want to switch to beekeeper

To cater the first it's easy we just have to add the annotation to newly created tables.
For the second we could say the user needs tol add the annotation manually (I am leaning towards this). Or we go add something in CT that adds the annotation always when doing an update.

Reply:
I don't mind, depends on what the user would expect to happen/making sure we document it clearly. As it stands:

1. We are adding parameters to new and existing table replications
2. If we want to change this to only add the params on new table replications, the annotation here should be removed, and I will need to make a change `Replica` to only add the annotations if the table has been newly created.

I'll make the change so we can see what this looks like.

What do we all think?